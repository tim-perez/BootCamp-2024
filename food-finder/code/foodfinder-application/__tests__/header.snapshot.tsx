import { act, render } from "@testing-library/react";
import { useSession } from "next-auth/react";
import Header from "components/header";

jest.mock("next-auth/react");
describe("The Header component", () => {
    it("renders unchanged when logged out", async () => {
        (useSession as jest.Mock).mockReturnValueOnce({
            data: { user: {} },
            status: "unauthenticated",
        });
        let container: HTMLElement | undefined = undefined;
        await act(async () => {
            container = render(<Header />).container;
        });
        expect(container).toMatchSnapshot();
    });

    it("renders unchanged when logged in", async () => {
        (useSession as jest.Mock).mockReturnValueOnce({
            data: {
                user: {
                    name: "test user",
                    fdlst_private_userId: "rndmusr",
                },
            },
            status: "authenticated",
        });
        let container: HTMLElement | undefined = undefined;
        await act(async () => {
            container = render(<Header />).container;
        });
        expect(container).toMatchSnapshot();
    });
});
