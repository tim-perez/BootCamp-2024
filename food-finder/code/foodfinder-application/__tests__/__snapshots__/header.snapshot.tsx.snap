// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders unchanged when logged in 1`] = `
<div>
  <header
    class="root"
  >
    <div
      class="layout-grid"
    >
      <a
        class="root"
        href="/"
      >
        <img
          alt="Logo: Food Finder"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
      <span
        class="name"
      >
        Hi 
        <b>
          test user
        </b>
      </span>
      <nav
        class="root"
      >
        <div
          class="root  outline"
        >
          <span
            class="span"
          >
            <a
              href="/list/rndmusr"
            >
              Your wish list
            </a>
          </span>
        </div>
        <div
          class="root  blue"
        >
          <span
            class="span"
          >
            Sign out
          </span>
        </div>
      </nav>
    </div>
  </header>
</div>
`;

exports[`The Header component renders unchanged when logged out 1`] = `
<div>
  <header
    class="root"
  >
    <div
      class="layout-grid"
    >
      <a
        class="root"
        href="/"
      >
        <img
          alt="Logo: Food Finder"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
      <nav
        class="root"
      >
        <div
          class="root  blue"
        >
          <span
            class="span"
          >
            Sign in
          </span>
        </div>
      </nav>
    </div>
  </header>
</div>
`;
